daemon/call.c

```diff
@@ -208,6 +205,12 @@ static void call_timer_iterator(void *key, void *val, void *ptr) {
 		goto delete;
 	}
 
+	// other timeouts not applicable to foreign calls
+	if (IS_FOREIGN_CALL(c)) {
+		c->deleted = c->ml_deleted = poller_now + cm->conf.delete_delay;
```

Comment:
The idea behind this was: Foreign calls do not have media in normal operation mode (no redundancy fail-over scenario). In this case, we didn't want _rtpengine_ to delete this foreign calls due to _TIMEOUT_ or _SILENT_TIMEOUT_. Marking foreign calls as deleted seems to be from a time before we've introduce the _foreign_ flag. Now I think it should be enough to just check if a call is a foreign one or not, so no need marking foreign calls as deleted. We'll update the pull request. Ideas are welcome.